docs/detections/rules-ui-create.asciidoc

```diff
@@ -355,6 +355,10 @@ also affect this rule.
 .. *Building block* (optional): Select to create a building-block rule. By
 default, alerts generated from a building-block rule are not displayed in the
 UI. See <<building-block-rule>> for more information.
+.. *Indicator prefix override*: Specify the default threat indicator path that the indicator match rule will use when obtaining threat indicator data from the defined indicator indices. This data is used to enrich indicator match alerts with metadata about matched threat indicators. The default value for this field is `threat.indicator`. 
```

Comment:
This is more a question than a suggestion, but I would have thought it would be "threat indicator index" not "threat indicator path", but my searches didn't turn up many references to either one. Why is it threat indicator path instead of index?

Reply:
@benironside, my understanding is that this setting tells the rule the _path location_ of threat indicator fields. So, when the rule queries the indicator indices--which are specified when you create the rule (see step 1c [here](https://www.elastic.co/guide/en/security/master/rules-ui-create.html#create-indicator-rule))-- it knows how to identify fields with indicator data. @ecezalp please correct me if I'm wrong about any of this or missing anything!

Would it be more clear if I added `location` after `default threat indicator path` and changed `obtaining` to `searches for`?

Reply:
the value of `indicator_index` is used to determine which indices will be queried to find the indicator documents.
the value of `threat_indicator_path` is used to point out where the enrichment data is in a single indicator document.

so in ECS 1.11 a single incidator event roughly looks like what's below.
```
{
  agent: {name: "ece's computer"},
  event: {ingested: "this-date", category: "indicator"},
  threat: {indicator: {url: {full: "www.asdf.com"}, file: {hash: {md5: "asdfasd", sha1: "asdfasgsdfg"}}}},
  @timestamp: "timestampvalue"
}
```
then we have an event we want to enricht that look like 
```
{
  threat: enrichments: []
}
```
we find `threat_indicator_path` on the indicator, which is `threat.indicator`, which gives us the enrichment: `{url: {full: "www.asdf.com"}, file: {hash: {md5: "asdfasd", sha1: "asdfasgsdfg"}` there.

once the event is enriched with that  it looks like this
```
{
  threat: enrichments: [{matched: {id:"asdf", value: "asdf", type: "asdf"}, indicator: {ENRICHMENT_GOES_HERE }}]
}
```


so an enriched event will have `{url: {full: "www.asdf.com"}, file: {hash: {md5: "asdfasd", sha1: "asdfasgsdfg"}}` placed into `threat.enrichments[0].indicator`. if event already had another enrichment then it'll go to `threat.enrichments[1].indicator`

before indicator data had different mappings and it looked like the following, so we used `threatintel.indicator` as the `threat_indicator_path`
```
{
  agent: {name: "ece's computer"},
  event: {ingested: "this-date", category: "indicator"},
  threatintel: {indicator: {url: {full: "www.asdf.com"}, file: {hash: {md5: "asdfasd", sha1: "asdfasgsdfg"}}}},
  @timestamp: "timestampvalue"
}
```

for something custom that users may want to use threats could look like anything
```
{
  random: {extrarandom: {url: {full: "www.asdf.com"}, file: {hash: {md5: "asdfasd", sha1: "asdfasgsdfg"}}}},
  @timestamp: "timestampvalue"
}
```
than the threat_indicator_path should configured to be `random.extrarandom`. if user uses non ECS fields and try to read `threat.indicator` as the threat_indicator_path it might just resolve to `undefined` so there will be no data enriched on the event.

Reply:
I think I had a similar reaction to @benironside -- when I see "path" I usually think of a file path, or a URL path. Thank you @ecezalp for the explanation! Now I understand (i think?) we're talking about a path through the data structure within the indicator event's data. How about something like:
```suggestion
.. *Indicator prefix override*: Specify the default threat indicator data structure path that the indicator match rule will use to locate threat indicator data in the defined indicator indices. This data is used to enrich indicator match alerts with metadata about matched threat indicators. The default value for this field is `threat.indicator`. 
```

Reply:
I like the suggestion to specify that we're talking about the data structure--"path" is pretty vague and can be even more so for newer users. Thanks all for the input!